storage/trace_test.go

```diff
@@ -16,251 +16,142 @@ package storage
+	otcodes "go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/sdk/instrumentation"
-	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/sdk/trace/tracetest"
-	"go.opentelemetry.io/otel/trace"
-	"google.golang.org/api/option"
+	"google.golang.org/api/googleapi"
 )
 
-type emulatedTraceTest struct {
-	*testing.T
-	resources resources
-}
-
-func TestTraceStorageTraceStartEndSpan(t *testing.T) {
+func TestStorageTraceStartEndSpan(t *testing.T) {
 	ctx := context.Background()
-	e := tracetest.NewInMemoryExporter()
-	tp := sdktrace.NewTracerProvider(sdktrace.WithSyncer(e))
-	defer tp.Shutdown(ctx)
-	otel.SetTracerProvider(tp)
+	te := testutil.NewOpenTelemetryTestExporter()
+	t.Cleanup(func() {
+		te.Unregister(ctx)
+	})
 
 	// TODO: Remove setting development env var upon launch.
 	t.Setenv("GO_STORAGE_DEV_OTEL_TRACING", "true")
 
-	spanName := "storage.TestTrace.TestStorageTraceStartEndSpan"
-	addAttrs := attribute.String("foo", "bar")
-	spanStartOpts := []trace.SpanStartOption{
-		trace.WithAttributes(addAttrs),
-	}
+	spanName := "storage.TestTrace.TestStartEndSpan"
+	ctx, span := startSpan(ctx, spanName)
 	newAttrs := attribute.Int("fakeKey", 800)
-
-	ctx, span := startSpan(ctx, spanName, spanStartOpts...)
 	span.SetAttributes(newAttrs)
 	endSpan(ctx, nil)
 
-	spans := e.GetSpans()
+	spans := te.Spans()
+	gotSpan := spans[0]
 	if len(spans) != 1 {
 		t.Errorf("expected one span, got %d", len(spans))
 	}
+	if got, want := gotSpan.Name, appendPackageName(spanName); got != want {
+		t.Fatalf("got %s, want %s", got, want)
+	}
 
-	// Test StartSpanOption and Cloud Trace adoption common attributes are appended.
-	// Test startSpan returns the span and additional attributes can be set.
-	wantSpan := createWantSpanStub(spanName)
-	wantSpan.Attributes = append(wantSpan.Attributes, addAttrs)
+	wantSpan := createWantSpanStub(spanName, getCommonAttributes())
 	wantSpan.Attributes = append(wantSpan.Attributes, newAttrs)
 	opts := []cmp.Option{
 		cmp.Comparer(spanAttributesComparer),
 	}
-	for _, span := range spans {
-		if diff := testutil.Diff(span, wantSpan, opts...); diff != "" {
-			t.Errorf("diff: -got, +want:\n%s\n", diff)
-		}
+	if diff := testutil.Diff(gotSpan, wantSpan, opts...); diff != "" {
+		t.Errorf("diff: -got, +want:\n%s\n", diff)
 	}
-	e.Reset()
 }
+func TestStorageTraceStartSpanOption(t *testing.T) {
+	ctx := context.Background()
+	te := testutil.NewOpenTelemetryTestExporter()
+	t.Cleanup(func() {
+		te.Unregister(ctx)
+	})
+
+	// TODO: Remove setting development env var upon launch.
+	t.Setenv("GO_STORAGE_DEV_OTEL_TRACING", "true")
+
+	spanName := "storage.TestTrace.TestStartSpanOption"
+	attrMap := make(map[string]interface{})
+	attrMap["my_string"] = "my string"
+	attrMap["my_bool"] = true
+	attrMap["my_int"] = 123
+	attrMap["my_int64"] = int64(456)
+	attrMap["my_float"] = 0.9
+	spanStartOpts := makeSpanStartOptAttrs(attrMap)
 
-func TestTraceSpansMultiEmulated(t *testing.T) {
```

Comment:
Why is this test removed?

Reply:
We can replace the emulated test with direct span checks. This change meets the same testing objectives and allows us to focus more directly on verifying the instrumentation and the bucket and object-specific attributes (upcoming in https://github.com/googleapis/google-cloud-go/commit/9701176fa0ad189ce7dd449af3f4424bcf05655b).

We've implemented granular tests for startSpan, endSpan, StartSpanOption, and error recording. The next PR also adds dedicated bucket and object trace tests.